internal/archive/archive.go

```diff
@@ -112,18 +120,38 @@ func (a *ubuntuArchive) selectPackage(pkg string) (control.Section, *ubuntuIndex
 	return selectedSection, selectedIndex, nil
 }
 
-func (a *ubuntuArchive) Fetch(pkg string) (io.ReadCloser, error) {
+func (a *ubuntuArchive) Fetch(pkg string) (io.ReadCloser, *PackageInfo, error) {
 	section, index, err := a.selectPackage(pkg)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	suffix := section.Get("Filename")
 	logf("Fetching %s...", suffix)
 	reader, err := index.fetch("../../"+suffix, section.Get("SHA256"), fetchBulk)
+	if err != nil {
+		return nil, nil, err
+	}
+	info := &PackageInfo{
+		Name:    section.Get("Package"),
+		Version: section.Get("Version"),
+		Arch:    section.Get("Architecture"),
+		SHA256:  section.Get("SHA256"),
+	}
+	return reader, info, nil
```

Comment:
Thanks, this looks good. Can we please have a `sectionPackageInfo` function to de-dup the code above?  We have that exact construction twice here and below.